carrot-store-UI/details/details.js

```diff
@@ -0,0 +1,88 @@
+function getDataFromServer(productId) {
+    fetch("http://localhost:3000/products/" + productId)
+        .then(displayResponse)
+        .then(displayData);
+}
+
+
+// logic to get de id from query param
+function getParameterByName(name, url) {
+    if (!url) url = window.location.href;
+    name = name.replace(/[\[\]]/g, '\\$&');
+    var regex = new RegExp('[?&]' + name + '(=([^&#]*)|&|#|$)'),
+        results = regex.exec(url);
+    if (!results) return null;
+    if (!results[2]) return '';
+    return decodeURIComponent(results[2].replace(/\+/g, ' '));
+}
+
+
+console.log("query param: " + window.location.search)
+
+
+productId = getParameterByName("id", window.location.search);
+console.log("productId is " + productId);
+
+if (productId === null) {
+    productId = "0"
+} 
+
+getDataFromServer(productId);
+
+function displayResponse(response) {
+    return response.json();
+}
+
+function displayData(data) {
+    console.log(data);
+    getProductDetails(data);
+
+    for (let i = 0; i < data.comments.length; i++) {
+        let user = data.comments[i];
+        getUsersData(user);
+    }
```

Comment:
It is good practice to extract a deeply nested value in a variable if we need to access it many times (e.g. for loop)

**const deeplyNestedVariable = a.b.c.d.e.f
for (let i = 0; i < deeplyNestedVariable ; i++) {...}**

That way we do not have to look through the object chain every loop.

We could also inline the user variable, e.g. :

**getUsersData(data.comments[i]);**